### ISSUE TITLE
SimpleScreenRecorder indicator not displaying in Ubuntu 17.10

### ISSUE THREAD

**Issue by Ads20000** [MaartenBaert/ssr#566]
See https://plus.google.com/+PopescuSorin/posts/HhSFUy8Cqsf (will link this Issue there), is it possibly because `XDG-CURRENT-DESKTOP` is set as `Unity` and should be set as `Unity:Unity7` as per [this post](https://didrocks.fr/2017/08/23/ubuntu-gnome-shell-in-artful-day-7/)? I can't find `XDG-CURRENT-DESKTOP` in the code though...

**Comment by MaartenBaert** [comment]
Ubuntu is doing some nonstandard stuff with the system tray. It's implemented by a Qt extension which translates the Qt systray API to the appindicator API which Ubuntu prefers. I did not implement this extension, it is just something that gets loaded automatically when you run SSR (or any other Qt program) on Ubuntu.

XDG_CURRENT_DESKTOP is an environment variable. You can see the current value in a terminal like this:
```
echo $XDG_CURRENT_DESKTOP
```
You can change it like this:
```
export XDG_CURRENT_DESKTOP="Unity:Unity7"
```
The change only applies to the current tab of the current terminal. You can run SSR from that terminal like this, to see if it fixes the problem:
```
simplescreenrecorder
```
If you want to make the change permanent, you should add the following line to your ```~/.profile``` file:
```
export XDG_CURRENT_DESKTOP="Unity:Unity7"
```
After that you need to log out and log in again for the change to take effect.

**Comment by Ads20000** [comment]
Yeah sorry I should've commented quicker but wanted Popescu to comment for herself rather than speaking for her (some people would have something to say about that...) :/
I'm not sure this is the actual problem because it does work on Ubuntu 17.04 and the `XDG_CURRENT_DESKTOP` bug applies to 17.04 too but this bug is not occurring on 17.04. I've asked Popescu to file a downstream bug (Launchpad will pick up details about her system that may be useful) and then link it here.

**Comment by EnforcerZhukov** [comment]
I'm having the same problem under Ubuntu Mate 16.04, but the last time I used SSR the icon was showing OK!

![image](https://user-images.githubusercontent.com/17401854/31818090-e573e87e-b596-11e7-8644-4def6f9327db.png)

This is what I get when I run SSR from terminal:

```
david@Asus-D541S:~$ simplescreenrecorder
==================== SSR iniciado ====================
SimpleScreenRecorder 0.3.8
Compiled with GCC 5.4.0
Qt: header 4.8.7, lib 4.8.7
libavformat: header 56.40.101, lib 56.40.101
libavcodec: header 56.60.100, lib 56.60.100
libavutil: header 54.31.100, lib 54.31.100
libswscale: header 3.1.101, lib 3.1.101
[CPUFeatures::Detect] Características de la CPU: mmx sse sse2 sse3 ssse3 sse4_1 sse4_2
[ALSAInput::GetSourceList] Generating source list ...
[ALSAInput::GetSourceList] Found card: [hw:0] HDA Intel PCH
[ALSAInput::GetSourceList] Found device: [hw:0,0] HDA Intel PCH: ALC256 Analog
[PulseAudioInput::GetSourceList] Generating source list ...
[SourceNamesCallback] Found source: [alsa_output.pci-0000_00_1b.0.analog-stereo.monitor] Monitor of Audio Interno Estéreo analógico
[SourceNamesCallback] Found source: [alsa_input.pci-0000_00_1b.0.analog-stereo] Audio Interno Estéreo analógico
QSystemTrayIcon::setVisible: No Icon set
```

**Comment by MaartenBaert** [comment]
I don't think this is something I can fix in SSR, it's either a bug in the Ubuntu systray plugin, or a system configuration problem that is preventing SSR from finding the icons. This can happen with icon themes that forget to include a fallback to the 'hicolor' theme when they don't have a custom icon.

**Comment by Ads20000** [comment]
Also applies to Ubuntu 17.10 (with its modified GNOME session)

**Comment by EnforcerZhukov** [comment]
Installed Unity desktop on my Ubuntu Mate 16.04 and now it's not missing, but shows some sort of weird icon:
![image](https://user-images.githubusercontent.com/17401854/31875337-0cf0700a-b7ce-11e7-8bd7-fd4a4088b86b.png)
Tried switching to other icon themes with no luck. Weird thing is It was working fine before, maybe something got updated on the repositories.

**Comment by Ads20000** [comment]
@MaartenBaert 

@jhasse, 'AppIndicator/KStatusNotifierItem support for GNOME Shell' developer [says that this is a bug in Qt, not in Ubuntu](https://github.com/ubuntu/gnome-shell-extension-appindicator/issues/100#issuecomment-344592473). I notice he's left a comment on a Qt bug about this so hopefully they'll respond...that was marked as a dupe though and the bug it was marked as a duplicate of is marked fixed...
